pkg/providers/instancetype/instancetype.go

```diff
@@ -37,39 +37,36 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 	"knative.dev/pkg/logging"
 
+	"github.com/aws/karpenter-provider-aws/pkg/providers/amifamily"
 	"github.com/aws/karpenter-provider-aws/pkg/providers/pricing"
 	"github.com/aws/karpenter-provider-aws/pkg/providers/subnet"
 
 	"sigs.k8s.io/karpenter/pkg/cloudprovider"
 	"sigs.k8s.io/karpenter/pkg/utils/pretty"
-
-	"github.com/aws/karpenter-provider-aws/pkg/providers/amifamily"
-)
-
-const (
-	InstanceTypesCacheKey         = "types"
-	InstanceTypeOfferingsCacheKey = "offerings"
 )
 
 type Provider interface {
 	LivenessProbe(*http.Request) error
-
 	List(context.Context, *corev1beta1.KubeletConfiguration, *v1beta1.EC2NodeClass) ([]*cloudprovider.InstanceType, error)
+	UpdateInstanceTypes(ctx context.Context) error
+	UpdateInstanceTypeOfferings(ctx context.Context) error
 }
 
 type DefaultProvider struct {
 	region          string
 	ec2api          ec2iface.EC2API
 	subnetProvider  subnet.Provider
 	pricingProvider pricing.Provider
-	// Has one cache entry for all the instance types (key: InstanceTypesCacheKey)
-	// Has one cache entry for all the zones for each subnet selector (key: InstanceTypesZonesCacheKeyPrefix:<hash_of_selector>)
-	// Values cached *before* considering insufficient capacity errors from the unavailableOfferings cache.
+
+	// Values stored *before* considering insufficient capacity errors from the unavailableOfferings cache.
 	// Fully initialized Instance Types are also cached based on the set of all instance types, zones, unavailableOfferings cache,
 	// EC2NodeClass, and kubelet configuration from the NodePool
 
-	mu    sync.Mutex
-	cache *cache.Cache
+	muInstanceType          sync.RWMutex
```

Comment:
Typically, the thing that the mutex is locking on is what you put the mutex over, just by contention